### PR TITLE
Add reference number to search

### PR DESCRIPTION
**Pull request by slawosz**
Cosmetic product can be searched using UKCP number. https://regulatorydelivery.atlassian.net/jira/software/projects/COSBETA/boards/24?selectedIssue=COSBETA-1165

### CODE REVIEW

File: cosmetics-web/spec/features/notification_search_spec.rb [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+RSpec.feature "Search", :with_stubbed_mailer, :with_stubbed_notify, :with_2fa, :with_2fa_app, type: :feature do
   let(:user) { create(:poison_centre_user, :with_sms_secondary_authentication) }
 
-  let(:notification1) { create(:notification, :registered, :with_component, notification_complete_at: 1.day.ago, product_name: "Cream") }
-  let(:notification2) { create(:notification, :registered, :with_component, notification_complete_at: 2.days.ago, product_name: "Shower Bubbles") }
-  let(:notification3) { create(:notification, :registered, :with_component, notification_complete_at: 3.days.ago, product_name: "Bath Bubbles", category: :face_care_products_other_than_face_mask) }
+  let(:cream) { create(:notification, :registered, :with_component, notification_complete_at: 1.day.ago, product_name: "Cream") }
+  let(:shower_bubbles) { create(:notification, :registered, :with_component, notification_complete_at: 2.days.ago, product_name: "Shower Bubbles") }
+  let(:bath_bubbles) { create(:notification, :registered, :with_component, notification_complete_at: 3.days.ago, product_name: "Bath Bubbles", category: :face_care_products_other_than_face_mask) }
```
Comment (scruti) [inline cosmetics-web/spec/features/notification_search_spec.rb:9]: Easier to track around. Nice.

**Review by scruti — APPROVED** [review]
Nice!

**Review by nasirkhanbeis — COMMENTED** [review]
When I try search by full number eg. 'UKCP-10513701' it is returning all the notifications with the searched on the top. I think it should return only 1 row. looks like it is considering UKCP which is matching all the rows

**Review by nasirkhanbeis — APPROVED** [review]
Happy with current implementation. please note, when user search by UKCP prefix, it returns exact match on the top + returns rest all other rows,. Agreed to keep as is and when Bruce around we will check if we need to any further improvements.

**Pull request merged**